src/main/java/seedu/address/logic/commands/CopyCommand.java

```diff
@@ -0,0 +1,181 @@
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Copies one detail of the prefix identified\n"
+        + "Parameters: "
+        + "[" + PREFIX_NAME + "] "
+        + "[" + PREFIX_PHONE + "] "
+        + "[" + PREFIX_EMAIL + "] "
+        + "[" + PREFIX_ADDRESS + "] "
+        + "[" + PREFIX_TAG + "] "
+        + "[" + PREFIX_ASSET + "]\n"
+        + "Example: " + COMMAND_WORD + " 1 "
+        + PREFIX_NAME + " ";
+
+    public static final String MESSAGE_NO_PARAM = "One field to copy must be provided.";
+    public static final String MESSAGE_EXTRA_PARAM = "Only one field can be copied.";
+
+    private final Index index;
+    private final boolean[] info;
+
+    /**
+     * @param index of the person in the filtered person list to copy
+     * @param info list of details to copy to clipboard
+     */
+    public CopyCommand(Index index, boolean[] info) {
+        requireNonNull(index);
+
+        this.index = index;
+        this.info = info;
+    }
+
+    @Override
+    public String execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToCopy = lastShownList.get(index.getZeroBased());
+        String personDetails = copyToClipboard(personToCopy, info);
+
+        return String.format(Messages.MESSAGE_COPIED, personDetails);
+    }
+
+    /**
+     * Copies the details of {@code personToCopy} to clipboard
+     */
+    private static String copyToClipboard(Person personToCopy, boolean[] info) {
+        requireNonNull(personToCopy);
+
+        if (info[0]) {
+            return personToCopy.getName().toString();
+        }
+        if (info[1]) {
+            return personToCopy.getPhone().toString();
+        }
+        if (info[2]) {
+            return personToCopy.getEmail().toString();
+        }
+        if (info[3]) {
+            return personToCopy.getAddress().toString();
+        }
+        if (info[4]) {
+            return personToCopy.getTags().toString();
+        }
+        if (info[5]) {
+            return personToCopy.getAssets().toString();
+        }
+
+        throw new IllegalArgumentException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, MESSAGE_USAGE));
+    }
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the CopyCommand
+     * and returns an CopyCommand object for execution.
+     * @throws IllegalArgumentException if the user input does not conform the expected format
+     */
+    public static CopyCommand of(String args) throws IllegalArgumentException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL,
+                PREFIX_ADDRESS, PREFIX_TAG, PREFIX_ASSET);
```

Comment:
Would it be better to combine these 2 lines?

Reply:
combined, and left new line at 8 spaces